NEXT-12716 - Add guide on how to remove a javascript plugin (#132)
shopware/docs: guides/plugins/plugins/storefront/remove-unnecessary-js-plugin.md

# Remove Javascript plugin

## Overview

When you develop your own plugin, you might want to exclude Javascript plugins at some occasions. For example, if you 

> don't want a Core plugin to interfere, with your own code. This guide will teach you how to remove this Javascript plugin from
> your own Shopware plugin.

Reviewer: ```suggestion
don't want a Core plugin to interfere, with your own code. This guide will teach you how to remove this Javascript plugin with
your own Shopware plugin.
```

> # Remove unnecessary Javascript plugin

Reviewer 2: Tbh I'm not sure about the "unnecessary". To me it sounds a bit have you have to remove a plugin which was unnecessary in the first place. Maybe just "Remove a Javascript plugin"

## Overview

When you develop your own plugin, you might want to exclude Javascript plugins at some occasions. For example, if you 

> don't want a Core plugin to interfere. This guide will teach you how to remove the unnecessary Javascript plugin from

Reviewer 3: ```suggestion
don't want a Core plugin to interfere, with your own code. This guide will teach you how to remove the unnecessary Javascript plugin from
```

> your one plugin.

Reviewer 4: ```suggestion
your own plugin.
```

## Prerequisites

> While this is not mandatory, knowing the guide about [adding custom javascript plugins](./add-custom-javascript.md) 
> beforehand might help you understand this guide a bit further. 

Reviewer 3: ```suggestion
While this is not mandatory, having read the guide about [adding custom javascript plugins](./add-custom-javascript.md) beforehand might help you understand this guide a bit further. 
```

Other than that, this guide just requires you to have a running plugin installed, e.g. our plugin from the 
[plugin base guide](./../plugin-base-guide.md).

## Unregistering Javascript Plugin

> Imagine we want to exclude the `OffcanvasCart` plugin, just to get a test case which can be inspected easily.

Reviewer 4: ```suggestion
Imagine we want to exclude the `OffCanvasCart` plugin, just to get a test case which can be inspected easily.
```

In order to remove an unnecessary plugin, you only need to add the following line to your `main.js` file:

> {% code title="<plugin root>/<plugin root>src/Resources/app/storefront/src/main.js" %}

Reviewer 3: ```suggestion
{% code title="<plugin root>/src/Resources/app/storefront/src/main.js" %}
```

```javascript
window.PluginManager.deregister('OffCanvasCart', '[data-offcanvas-cart]');
```
{% endcode %}

> After building the storefront anew, you shouldn't be able to open the offcanvas cart anymore. Another useful way of
> testing the removal of the unnecessary plugin is using your browser's devtools. Just open your devtool's console and 

Reviewer 3: ```suggestion
After building the storefront anew, you shouldn't be able to open the offcanvas cart anymore. Another useful way of
testing this is using your browser's devtools. Just open your devtool's console and 
```

Reviewer 3: U don't have to apply this if u dont want to

type in `PluginManager.getPluginList()` in order to get a list of all registered plugins.

> In our case, we shouldn't find `OffcanvasCart` in the listed plugins anymore.

Reviewer 4: ```suggestion
In our case, we shouldn't find `OffCanvasCart` in the listed plugins anymore.
```
